### PR TITLE
fix: validation of dynamodb attribute name

### LINKED ISSUE #2134: Storage init fails due to validateKey
When creating DynamoDB with the following options(the length of attribute name is 2):
`$ copilot storage init -n todotable -t "DynamoDB" -w backend --partition-key Id:S --no-lsi --no-sort`

Copilot throws the following error:
`✘ value must contain only alphanumeric characters and ._-`

If the length of DynamoDB table's attribute name is longer like following:
`$ copilot storage init -n todotable -t "DynamoDB" -w backend --partition-key TodoId:S --no-lsi --no-sort`

Copilot wrote CFn template successfully.
`✔ Wrote CloudFormation template for DynamoDB Table todotable at copilot/backend/addons/todotable.yml`

According to DynamoDB's naming rule(https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.NamingRulesDataTypes.html#HowItWorks.NamingRules ), attribute names must be at least one character long. 

Validation for the length of partition key name is needed in validateKey function
https://github.com/aws/copilot-cli/blob/d5d54a5f9583b74f0904b1a7954399ba04d65b9e/internal/pkg/cli/validate.go#L515

### PR DESCRIPTION
**Pull request by ghost**
Fixes #2134

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/validate.go [hunk 1]
```diff
@@ -521,9 +537,9 @@ func validateKey(val interface{}) error {
 	if err != nil {
 		return errDDBAttributeBadFormat
 	}
-	err = dynamoTableNameValidation(*attr.Name)
+	err = dynamoAttributeNameValidation(*attr.Name)
 	if err != nil {
-		return errValueBadFormatWithPeriodUnderscore
+		return errDDBAttributeBadSize
```
Comment (Lou1415926) [inline internal/pkg/cli/validate.go]: Maybe we can `return err` here since it's also possible that `err = errValueNotAString`

**Review by Lou1415926 — COMMENTED** [review]
Thank you so much for the fix!!! We really appreciate it 🙇🏼‍♀️

File: internal/pkg/cli/validate.go (hunk 1 above)
Comment (ghost) [reply internal/pkg/cli/validate.go]: Thank you for your review. I agree with you and made another commit.

**Review by Lou1415926 — APPROVED** [review]
Looks great! Thank you very much for the fix!!

File: internal/pkg/cli/validate.go [hunk 2]
```diff
@@ -521,9 +537,9 @@ func validateKey(val interface{}) error {
 	if err != nil {
 		return errDDBAttributeBadFormat
 	}
-	err = dynamoTableNameValidation(*attr.Name)
```
Comment (efekarakus) [inline internal/pkg/cli/validate.go:524]: Is `dynamoTableNameValidation` used anywhere? if not can we delete it as well?

Comment (Lou1415926) [reply internal/pkg/cli/validate.go:524]: I think it is still in use in storage_init.go to validate DDB table name

**Review by efekarakus — APPROVED** [review]

**Pull request merged**